### ISSUE TITLE
Bug in the month of April

### ISSUE THREAD

**Issue by WaelSleiman** [square/android-times-square#269]
To whom it may concern,
The issue is in the month of april. All the april days start one day after the last day of march.
Example: April 1, 2016 is being shown as Saturday while March 31, 2016 is a Thurday in the com.squareup.timessquare.CalendarPickerView.
This error is being repeated over the year.

Thank you for your time.

**Comment by edenman** [comment]
Screenshot?  I have no idea what you're saying.

**Comment by WaelSleiman** [comment]
Sorry for the delay.
![screenshot](https://cloud.githubusercontent.com/assets/18210544/14239378/ba611d24-fa43-11e5-8024-8dfca1bb4f64.png)

**Comment by edenman** [comment]
That is very strange.  Works fine for me: 
<img width="484" alt="screen shot 2016-04-04 at 11 42 51 am" src="https://cloud.githubusercontent.com/assets/1063557/14259147/6ed9dfbe-fa5a-11e5-9239-86ed004c49de.png">

Can you post the code you're using to initialize the view?  Also, what locale are you in?

**Comment by WaelSleiman** [comment]
After further testing on other devices, I found out that this bug is not appearing on my Nexus 6 
while it is still happening on my HTC one M9 and Samsung J5.
The local being used by all three is GMT +03:00 Eastern European Summer Time
The native date is working as it should.
All three are using the same project build.

As for the initialization of the calendar,
private Date minDate;
    private Date maxDate;
CalendarPickerView calendarView;

minDate = Calendar.getInstance();
minDate.setTime(date);
minDate.add(Calendar.DAY_OF_YEAR, -1);

maxDate = Calendar.getInstance();
maxDate.setTime(date);
maxDate.add(Calendar.YEAR, 1);

date = minDate.getTime();

calendarView.init(minDate.getTime(), maxDate.getTime()).inMode(CalendarPickerView.SelectionMode.SINGLE).withSelectedDate(date);

**Comment by edenman** [comment]
Well that's perplexing.  I can't reproduce the issue on any of the phones I have at my disposal so I'm not going to be able to help.  Try running the sample app on one of your phones to see if it has the same issue?  I'd love a PR if you figure out where the bug is.

**Comment by WaelSleiman** [comment]
I checked using the sample app and received the same result as my app.
Thank you for your time

**Comment by edenman** [comment]
Hrm.  If you see the same issue in the sample app, that definitely sounds like a bug...maybe hook up the debugger and see if you can see why it's selecting the wrong date?

**Comment by WaelSleiman** [comment]
Hello again,
So i was able to find the cause of this bug.
Apparently i was mistaken in my previous statement when i told you that this bug does not appear on my Nexus device. I was able to reproduce it by changing locales.
To reproduce it, change the local to Beirut(+3:00). I don't know if it exist in other locales. 
Here is a video to view the bug in action:
https://www.youtube.com/watch?v=0r4GTM98Ndo&feature=youtu.be

**Comment by edenman** [comment]
Innnnteresting.  I don't have time to look at this right now, but definitely curious to see what's busted here.  PR welcome if you figure it out!

**Comment by cnordvik** [comment]
We have a user with the same bug (from Libanon). Notice that April 1 is both on Saturday and Sunday here. His feedback was: 

> April 1 starts on Sunday instead of Saturday so whenever we click on Wednesday 4, for example, it redirects us to Tuesday 4

![pasted image at 2017_03_29 10_23](https://cloud.githubusercontent.com/assets/584549/24445474/4b41a35e-146a-11e7-8052-12719f1a81af.png)

**Comment by Namiii** [comment]
We have a handful of reports of this happening as well, have there been any developments on this issue or nothing yet?

**Comment by edenman** [comment]
Nope.  I haven't been able to repro.  PRs welcome if you figure it out!

**Comment by akul1994** [comment]
happening with me as well. using a samsung galaxy s6 edge. any updates?

**Comment by ashish21** [comment]
Having the same issue for IST+5.30 timezone as well on some devices. @edenman any insight??

**Comment by jontelang** [comment]
I cannot provide a proper solution, but something of a workaround. Our app uses a fixed header where the order of the weekdays are hardcoded. We had the same issue, and our solution was to force the calendar to use a US locale. This fixed it on all our devices that had the issue.

It's not an exact solution, but it might help someone out.

We have not yet tried to reproduce the issue with the default 'weekday header', but we might do that later.

The code in our app looks like this:

    calendar.init(cal.getTime(), mCalendar.getTime(), new Locale(Locale.getDefault().getLanguage(), Locale.US.getCountry()))

_Note that I am not the Android developer, so the solution is somewhat paraphrased._

**Comment by kidsh3148** [comment]
@jontelang Yes, your solution will definitely help us for showing the dates on a proper day, but then when we are clicking on the particular date it will return the date before one day for the first time.

`calendar.setOnDateSelectedListener(new CalendarPickerView.OnDateSelectedListener() {
                                                   @Override
                                                   public void onDateSelected(final Date date) {`


If I am clicking on the 20th Jan 2019 then it will return me `Sun Jan 19 00:00:00 GMT-01:00 2019`.

Did you faced something like this!!??
